src/net_processing.cpp

```diff
@@ -3944,6 +4160,163 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
+
+    // Record an (expected) reconciliation request with parameters to respond when time comes.
+    // All initial reconciliation responses will be done at the same time to prevent tx-related privacy leaks.
+    if (msg_type == NetMsgType::REQRECON) {
+        LOCK(peer->m_recon_state_mutex);
+        if (peer->m_recon_state == nullptr) return;
+        if (!peer->m_recon_state->m_requestor) return;
+        if (peer->m_recon_state->m_incoming_recon != RECON_NONE) return;
```

Comment:
Question: is it possible for `peer->m_recon_state->m_outgoing_recon != RECON_NONE`, i.e. we sent them a `REQRECON` around the same time they sent us one?